modules/nw-control-dns-records-public-aws-with-VPC.adoc

```diff
@@ -0,0 +1,81 @@
+
+With VPC sharing, we will have two Amazon AWS accounts, one with a VPC and a Route 53 hosted zone configured (Account A), and the other for installing a cluster (Account B). We will configure ExternalDNS Operator in Account B to create records in a Route 53 hosted zone Account A. In this example, we will illustrate the process of sharing a private hosted zone.
+
+.Prerequisites
+* You have two Amazon AWS accounts, one with a VPC and a Route 53 private hosted zone configured (Account A), and the other for installing a cluster (Account B).
+* You have created an IAM Policy and IAM Role with the appropriate permission in Account A for Account B to create DNS records in Account A's Route 53.
+* You have installed a cluster in Account B into the existing VPC of Account A as documented in xref:../../installing/installing_aws/installing-aws-vpc.adoc#installing-aws-vpc[Installing a cluster on AWS into an existing VPC].
+* You have installed the ExternalDNS Operator in the cluster in Account B as documented in xref:../../networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc#installing-external-dns-on-cloud-providers[Installing External DNS Operator on cloud providers].
+
+.Procedure
+
+. Get the Role ARN of the IAM Role you created for allowing Account B to access Account A's Route 53:
++
+[source,terminal]
+----
+$ aws --profile account-a iam get-role --role-name user-rol1 | head -1
+----
++
+.Example output
++
+[source,terminal]
+----
+ROLE	arn:aws:iam::1234567890123:role/user-rol1	2023-09-14T17:21:54+00:00	3600	/	AROA3SGB2ZRKRT5NISNJN	user-rol1
+----
+
+. Locate the private hosted zone you wish to use with Account A's credentials:
++
+[source,terminal]
+----
+$ aws --profile account-a route53 list-hosted-zones | grep testextdnsoperator.apacshift.support
+----
++
+.Example output
++
+[source,terminal]
+----
+HOSTEDZONES	terraform	/hostedzone/Z02355203TNN1XXXX1J6O	testextdnsoperator.apacshift.support. 5
+----
+
+. Create the ExternalDNS object:
++
+[source,terminal]
+----
+$ cat <<EOF | oc create -f -
+apiVersion: externaldns.olm.openshift.io/v1beta1
+kind: ExternalDNS
+metadata:
+  name: sample-aws 
+spec:
+  domains:
+  - filterType: Include   
+    matchType: Exact   
+    name: testextdnsoperator.apacshift.support 
+  provider:
+    type: AWS
+    aws: 
+      assumeRole: arn:aws:iam::12345678901234:role/user-rol1 <1>
+      credentials:
+  source:  
+    type: OpenShiftRoute 
+    openshiftRouteOptions:
+      routerName: default 
+EOF
+----
+<1> Specify the Role ARN  to have DNS records created in Account A.
+
+. Check the records created for OCP routes by using the following command:
++
+[source,terminal]
+----
+$ aws --profile account-a route53 list-resource-record-sets --hosted-zone-id Z02355203TNN1XXXX1J6O --query "ResourceRecordSets[?Type == 'CNAME']" | grep console+
+----
```

Comment:
I recommend including example output following the verification command, if possible.

Reply:
@gcs278 do you have example output?

Reply:
Yes - I got it now, if you make the change I suggested in https://github.com/openshift/openshift-docs/pull/68016/files#r1399775161, it should be:
```
console-openshift-console.apps.gspence-demovpc2.devcluster.openshift.com.	300	CNAME
```

Reply:
Oh, one more minor point - if we add example output, should we be consistent by also adding the same example output here:
https://github.com/openshift/openshift-docs/blob/3770f73e0051c332dd18cb45f911e1e07a10e0d5/modules/nw-control-dns-records-public-hosted-zone-aws.adoc?plain=1#L96

This is the existing module that will be right above this new one (https://docs.openshift.com/container-platform/4.13/networking/external_dns_operator/nw-creating-dns-records-on-aws.html). I used this module as a template (hence why I didn't have example output). Might be good to be consistent since they are so similar.